src/uds/mod.rs

```diff
@@ -35,407 +32,116 @@ pub use read_dtc_information::*;
 pub use scaling_data::*;
 pub use security_access::*;
 
-/// FIXME: Use UdsCommand instead
-/// Note: `#[deprecated]` doesn't work here due to https://github.com/rust-lang/rust/issues/30827
-pub use auto_uds::UdsCommand as UDSCommand;
-
-/// FIXME: Use UdsError instead
-/// Note: `#[deprecated]` doesn't work here due to https://github.com/rust-lang/rust/issues/30827
-pub use auto_uds::UdsError as UDSError;
-
-pub use auto_uds::{UdsCommand, UdsError};
-use auto_uds::{UdsCommandByte, UdsErrorByte};
-
-fn lookup_uds_nrc(x: u8) -> String {
-    format!("{:?}", UdsErrorByte::from(x))
-}
+pub use auto_uds::{UdsError};
 
-#[derive(Debug, Copy, Clone)]
-#[repr(C)]
-/// UDS server options
-pub struct UdsServerOptions {
-    /// ECU Send ID
-    pub send_id: u32,
-    /// ECU Receive ID
-    pub recv_id: u32,
-    /// Read timeout in ms
-    pub read_timeout_ms: u32,
-    /// Write timeout in ms
-    pub write_timeout_ms: u32,
-    /// Optional global address to send tester-present messages to
-    /// Set to 0 if not in use
-    pub global_tp_id: u32,
-    /// Tester present minimum send interval in ms
-    pub tester_present_interval_ms: u32,
-    /// Configures if the diagnostic server will poll for a response from tester present.
-    pub tester_present_require_response: bool,
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum UDSErrorTy {
```

Comment:
I am not sure you need this abstraction - `auto_uds::UdsErrorByte` is the same enum, which is an alias to `ByteWrapper<UdsError>`

Reply:
The abstraction is needed as it seems UDSError doesn't like being created from any byte. Some ECUs respond with custom error codes fyi

Reply:
@rnd-ash I didn't say `UdsError`, I said `UdsErrorByte` -- every enum I have has a corresponding `*Byte` wrapper specifically designed to handle the "non-standard"  case.  See https://github.com/nyurik/auto_uds/blob/main/src/utils.rs

Reply:
Fixed in 7c3cfa9f1897f5ba55881d743a23f9087a7cd5a2